### ISSUE TITLE
Improve message when a panel is not supported by the current run-mode

### ISSUE THREAD

**Issue by johnpryan** [flutter/devtools#4050]
It's not clear what DevTools the user is allowed to open and when. For example, if a user is running in profile mode and tries to run the "Open Widget Inspector" command in VSCode they are shown this screen:

<img width="712" alt="Screen Shot 2022-04-27 at 12 54 48 PM" src="https://user-images.githubusercontent.com/1145719/165630692-832f881c-fc32-4a5c-8ef7-077ca0857ff7.png">

It seems like that most things are supported in Debug and Profile mode, but show inconsistent messages:

|  | Debug | Profile | Release |
|---|---|---|---|
| Widget Inspector | 🟢 | 🔴 | 🔴 |
| CPU Profiler | 🔴 | 🟢 | 🔴 |
| Memory Page | 🟡 | 🟢 | 🔴 |
| Performance Page | 🔴 | 🟢 | 🔴 |
| Network Page | 🟢 | 🟢 | 🔴 |
| Logging Page | 🟢 | 🟢 | 🔴 |

For example, when running in debug mode, the toast messages for the CPU Profiler, Memory, and Performances pages are inconsistent. The memory page displays a warning but the CPU Profiler and Performance pages display an error:

<img width="480" alt="Screen Shot 2022-04-27 at 2 05 46 PM" src="https://user-images.githubusercontent.com/1145719/165631655-0d3993f8-8fc7-45d0-b855-18dfcb3c3a60.png">

<img width="558" alt="Screen Shot 2022-04-27 at 2 08 35 PM" src="https://user-images.githubusercontent.com/1145719/165631689-ba95d4d1-a47c-4904-8c69-1b669b3bc58a.png">

If you try to open a DevTools panel in release mode, you get a warning like this, but I'm also wondering if we could clarify this message:

<img width="463" alt="Screen Shot 2022-04-27 at 2 06 48 PM" src="https://user-images.githubusercontent.com/1145719/165631917-8c92a7ba-a4ca-4345-8b21-2ed3588546e0.png">

cc: @kenzieschmoll @DanTup

**Comment by DanTup** [comment]
> If you try to open a DevTools panel in release mode, you get a warning like this, but I'm also wondering if we could clarify this message:
> 
> > This debug session is not ready yet or does not support DevTools.

Were you running on the web when you got this message? This message shows up when there's no VM Service URI. This usually means the session is still starting, but I believe for Release builds on web we'll never get one. We can tweak this text to be a bit more specific, although I've tried not to encode too many assumptions about the specific behaviour into VS Code where things might change in Flutter. This is also why some commands might still show up in VS Code, but then it turns out that page is not available (VS Code doesn't really know for certain whether you're running on web, for example - it would have to make assumptions about devices based on their IDs/names, including reading any custom `args` in your launch config).

Perhaps it would be useful to have an API (for ex. via the DevTools server?)  that would allow editors to query the capabilities for DevTools for a given session? Eg., given this VM Service URI, which DevTools pages will be available?

**Comment by johnpryan** [comment]
> Were you running on the web when you got this message? 

No, I was running on desktop. I think the widget inspector is unsupported in profile mode, so there's nothing for this panel to do.

**Comment by DanTup** [comment]
Ah, gotcha. I guess it makes sense we also don't have VM Service URIs for Desktop release builds.

Right now there's not an easy way for VS Code to know whether it doesn't have a VM Service URL because the debug session has not finished starting up, or it's a release build. I think we should be able to get this though (I can see the `flutter` tool does emit an `app.started` event in the case of a release build). I've filed https://github.com/Dart-Code/Dart-Code/issues/3945 to do this and improve that specific message which covers the case where DevTools won't be usable at all. I think it makes sense for the other cases - where we can spawn DevTools but that screen is not available for the current app - in DevTools by tweaking the message shown in the original screenshot at the top.

**Comment by DanTup** [comment]
With the latest Flutter and next VS Code extension, the message here:

![image](https://user-images.githubusercontent.com/1078012/175082803-4481b6dc-7755-42e7-8dd8-5f22c0a5aaaa.png)

Should be a bit more specific, separating "is not ready yet" from "is not available for an app running in this mode".
